packages/api/src/routes/medical/internal-cq.ts

```diff
@@ -110,13 +112,15 @@ router.get(
 // BELOW ARE THE ROUTES PERTAINING TO THE IHE-GATEWAY
 
 /**
- * POST /internal/carequality/patient-discovery/response
+ * POST /internal/carequality/patient-discovery/response/response
  *
  * Receives a Patient Discovery response from the IHE Gateway
  */
 router.post(
   "/patient-discovery/response",
   asyncHandler(async (req: Request, res: Response) => {
+    const pdResponse = patientDiscoveryResponseSchema.parse(req.body);
+    await createPatientDiscoveryResult(pdResponse);
     await handleIHEResponse({
```

Comment:
Doesn't it make sense to have `createPatientDiscoveryResult()` being called inside `handleIHEResponse()`?

Also, since we're adding a schema for the input, we could pass the typed to const to `handleIHEResponse` (`req.body` is `any`).